cluster-autoscaler/simulator/dynamicresources/utils/resource_claims.go

```diff
@@ -114,3 +115,37 @@ func PodClaimOwnerReference(pod *apiv1.Pod) metav1.OwnerReference {
 func claimConsumerReferenceMatchesPod(pod *apiv1.Pod, ref resourceapi.ResourceClaimConsumerReference) bool {
 	return ref.APIGroup == "" && ref.Resource == "pods" && ref.Name == pod.Name && ref.UID == pod.UID
 }
+
+// ClaimWithoutAdminAccessRequests returns a copy of the claim without admin access requests, and their results.
+func ClaimWithoutAdminAccessRequests(claim *resourceapi.ResourceClaim) *resourceapi.ResourceClaim {
+	claimCopy := claim.DeepCopy()
+	if claimCopy.Status.Allocation == nil {
+		return claimCopy
+	}
+	deviceRequestAllocationResults := make([]resourceapi.DeviceRequestAllocationResult, 0)
+	for _, deviceRequestAllocationResult := range claimCopy.Status.Allocation.Devices.Results {
+		// Device requests with AdminAccess don't reserve their allocated resources, and are ignored when scheuling.
+		devReq := getDeviceResultRequest(claim, &deviceRequestAllocationResult)
+		if ptr.Deref(devReq.Exactly.AdminAccess, false) {
+			continue
+		}
+		deviceRequestAllocationResults = append(deviceRequestAllocationResults, deviceRequestAllocationResult)
+	}
+	if claimCopy.Status.Allocation != nil {
```

Comment:
L122 already guarantees that `claimCopy` won't be nil at this point in the execution flow, so we can simply assign `deviceRequestAllocationResults` to `claimCopy.Status.Allocation.Devices.Results` without checking for nil here.